packages/@expo/cli/src/start/server/DevServerManager.ts

```diff
@@ -152,16 +153,26 @@ export class DevServerManager {
   }
 
   async bootstrapTypeScriptAsync() {
-    if (await this.ensureProjectPrerequisiteAsync(TypeScriptProjectPrerequisite)) {
+    const typescriptPrerequisite = await this.ensureProjectPrerequisiteAsync(
+      TypeScriptProjectPrerequisite
+    );
+
+    if (env.EXPO_NO_TYPESCRIPT_SETUP) {
       return;
     }
+
     // Optionally, wait for the user to add TypeScript during the
     // development cycle.
     const server = devServers.find((server) => server.name === 'metro');
     if (!server) {
       return;
     }
-    await server.waitForTypeScriptAsync();
+
+    if (!typescriptPrerequisite) {
+      server.waitForTypeScriptAsync();
```

Comment:
Does this mean the user needs to restart the process to enable route generation?